jobrunner/record_stats.py

```diff
@@ -0,0 +1,80 @@
+"""
+Super crude docker/system stats logger
+"""
+import datetime
+import json
+import logging
+import sqlite3
+import sys
+import time
+
+from . import config
+from .docker_stats import get_container_stats, get_volume_and_container_sizes
+from .system_stats import get_system_stats
+from .log_utils import configure_logging
+
+
+SCHEMA_SQL = """
+CREATE TABLE stats (
+    timestamp TEXT,
+    data TEXT
+);
+"""
+
+
+log = logging.getLogger(__name__)
+
+
+def main():
+    database_file = config.STATS_DATABASE_FILE
+    if not database_file:
+        log.info("STATS_DATABASE_FILE not set; not polling for system stats")
+        return
+    log.info(f"Logging system stats to: {database_file}")
+    connection = get_database_connection(database_file)
+    while True:
+        log_stats(connection)
+        time.sleep(config.STATS_POLL_INTERVAL)
+
+
+def get_database_connection(filename):
+    filename.parent.mkdir(parents=True, exist_ok=True)
+    conn = sqlite3.connect(filename)
+    # Enable autocommit
+    conn.isolation_level = None
+    schema_count = list(conn.execute("SELECT COUNT(*) FROM sqlite_master"))[0][0]
```

Comment:
This is nice an simple.

Also saw this nice way of doing native sqlite migrations - don't think we need it in this case, though

https://levlaz.org/sqlite-db-migrations-with-pragma-user_version/